cmake/onnxruntime.cmake

```diff
@@ -105,6 +105,7 @@ target_compile_definitions(onnxruntime PRIVATE VER_MINOR=${VERSION_MINOR_PART})
 target_compile_definitions(onnxruntime PRIVATE VER_BUILD=${VERSION_BUILD_PART})
 target_compile_definitions(onnxruntime PRIVATE VER_PRIVATE=${VERSION_PRIVATE_PART})
 target_compile_definitions(onnxruntime PRIVATE VER_STRING=\"${VERSION_STRING}\")
+target_compile_definitions(onnxruntime PRIVATE FILE_NAME=\"onnxruntime.dll\")
```

Comment:
Why was this change made? How does it relate to #11058 ?

Reply:
this change was needed because the onnxruntime.rc was hard coding onnxruntime.dll and we needed to change to ensure that filedetails for onnxruntime_providers_shared.dll were updated properly

Reply:
The file was created by Microsoft Windows team, and it was specific for that use. You can see it claims the file is a part of Windows.  But Windows does not ship onnxruntime_providers_shared.dll and the other DLLs.